src/DependencyInjection/DI/test/ServiceProviderContainerTests.cs

```diff
@@ -128,6 +128,33 @@ public void ResolvesServiceMixedServiceAndOptionalStructConstructorArguments()
             Assert.NotNull(service);
         }
 
+        [Fact]
+        public void ResolvesServiceMixedServiceAndOptionalStructConstructorArgumentsReliably()
+        {
+            var serviceCollection = new ServiceCollection();
+            serviceCollection.AddSingleton<IFakeService, FakeService>();
+            serviceCollection.AddTransient<ClassWithServiceAndOptionalArgsCtorWithStructs>();
+
+            var provider = CreateServiceProvider(serviceCollection);
+
+            for (int i = 0; i < 100; i++)
```

Comment:
Can you please also add a comment about why this loop is here

Reply:
Added